cogs/welcome/welcome.py

```diff
@@ -42,15 +52,91 @@ def __init__(self, bot: Red):
         self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
         self.config.register_guild(**DEFAULT_GUILD)
 
+    async def getRandomMessage(self, guild):
+        greetings = await self.config.guild(guild).greetings()
+        numGreetings = len(greetings)
+        if not greetings:
+            return "Welcome to the server {USER}"
+        else:
+            return random.choice(list(greetings.values()))
+
     # The async function that is triggered on new member join.
     @commands.Cog.listener()
     async def on_member_join(self, newMember: discord.Member):
+        await self.sendWelcomeMessageChannel(newMember)
         await self.sendWelcomeMessage(newMember)
 
     @commands.Cog.listener()
     async def on_member_remove(self, leaveMember: discord.Member):
         await self.logServerLeave(leaveMember)
 
+    # This async function is to look for if the welcome channel was removed
+    @commands.Cog.listener()
+    async def on_channel_remove(self, removedChannel: discord.TextChannel):
+        guild = removedChannel.guild
+        # the channel to post welcome stuff in
+        welcomeIDSet = await self.config.guild(guild).welcomeChannelSet()
+        welcomeID = await self.config.guild(guild).welcomeChannel()
+        # the channel msg should point to
+        welcomeLinkIDSet = await self.config.guild(guild).welcomeChannel_LINK_SET()
+        welcomeLinkID = await self.config.guild(guild).welcomeChannel_LINK()
+        if welcomeIDSet and removedChannel.id == welcomeID:
+            await self.config.guild(guild).welcomeChannelSet.set(False)
```

Comment:
Can we log that we set it to `False` if we enter this if block?

Reply:
Done I believe?